src/main/java/frc/robot/poseestimator/limelight/MultiLimelight.java

```diff
@@ -0,0 +1,111 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+
+public class MultiLimelight extends GBSubsystem {
+
+    private static MultiLimelight instance;
+
+    private List<Limelight> limelights;
+    private Pose2d currentPosition;
+
+    private MultiLimelight() {
+        super("MultiLimeLight/");
+        limelights = new ArrayList<>();
+        for (String limelightName : VisionConstants.LIMELIGHT_NAMES) {
+            limelights.add(new Limelight(limelightName));
+        }
+    }
+
+    public static void init() {
+        instance = new MultiLimelight();
+    }
+
+    public static MultiLimelight getInstance() {
+        if (instance == null) {
+            init();
+        }
+        return instance;
+    }
+
+    public List<Optional<Pair<Pose2d, Double>>> getAll2DEstimates() {
+        ArrayList<Optional<Pair<Pose2d, Double>>> estimates = new ArrayList<>();
+
+        for (Limelight limelight : limelights) {
+            currentPosition = Robot.getPosEstimator.getOdometryPose();
+            if (limelight.hasTarget()) {
+                if (getLimelightedOutputInTolerance(limelight)) {
+                    estimates.add(limelight.getUpdatedPose2DEstimation());
+                }
+            }
+        }
+
+        return estimates;
+    }
+
+    public boolean getLimelightedOutputInTolerance(Limelight limelight) {
+        Pose2d limelightPosition;
+        Transform2d transformDifference;
+        Rotation2d rotationDifference;
+
+        if (limelight.getAprilTagConfidence())
+            if (limelight.getUpdatedPose2DEstimation().isPresent()) {
+                limelightPosition = limelight.getUpdatedPose2DEstimation().get().getFirst();
+                transformDifference = limelightPosition.minus(currentPosition);
+                rotationDifference = limelightPosition.getRotation().minus(currentPosition.getRotation());
+
+                return transformDifference.getTranslation().getNorm() >= VisionConstants.POSITION_TOLERANCE
+                        && rotationDifference.getDegrees() >= VisionConstants.ROTATION_TOLERANCE.getDegrees();
+            }
+
+        return false;
+    }
+
+
+    public void recordEstimatedPositions() {
+        int i = 0;
+        for (Optional<Pair<Pose2d, Double>> estimation : MultiLimelight.getInstance().getAll2DEstimates()) {
+            if (estimation.isPresent()) {
+                Logger.recordOutput(super.getLogPath() + "estimation " + i, estimation.get().getFirst());
+            }
+        }
+    }
+
+    public double getDynamicStdDevs(int limelightId) {
+        return limelights.get(limelightId).getDistanceFromAprilTag() / VisionConstants.VISION_TO_STANDARD_DEVIATION;
+    }
+
+    public boolean hasTarget(int limelightId) {
+        return limelights.get(limelightId).hasTarget();
+    }
+
+    public Optional<Pair<Pose2d, Double>> getFirstAvailableTarget() {
+        for (Optional<Pair<Pose2d, Double>> output : getAll2DEstimates()) {
```

Comment:
same as above

Reply:
what same as above? Sorry but I don't understand what's wrong with that func